vpr/src/route/rr_graph_clock.h

```diff
@@ -0,0 +1,119 @@
+#ifndef RR_GRAPH_CLOCK_H
+#define RR_GRAPH_CLOCK_H
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <unordered_map>
+#include <set>
+#include <utility>
+
+#include "clock_fwd.h"
+
+#include "clock_network_types.h"
+#include "clock_connection_types.h"
+
+class ClockNetwork;
+class ClockConnection;
+
+class SwitchPoint {
+    public:
+        // [grid_width][grid_height][nodes]
```

Comment:
change [nodes] to [0..nodes_at_this_location-1]
Comment what this class is for, comment why you need rr_node_indices etc.
Comment what locations is. Sorted list of all the SwitchPoint locations so you can rapidly look up the locations where SwitchPoints exist.  Give example of how to use.  
Need big picture of how to use this class, maybe at the top of this file.
Include what rr_nodes are being built: e.g. spine, spine-to-rib-driver, rib.

Reply:
Done: commit 69d6e5dce3f2ed456f9cee8ef557abbee8f93d85